mixer/cmd/bundles.go

```diff
@@ -24,47 +23,91 @@ import (
 	"github.com/spf13/cobra"
 )
 
-type bundleCmdFlags struct {
-	all   bool
-	force bool
-	git   bool
-}
-
-var bundleFlags bundleCmdFlags
-
+// Top level bundle command ('mixer bundle')
 var bundleCmd = &cobra.Command{
 	Use:   "bundle",
 	Short: "Perform various actions on bundles",
 }
 
-var addBundlesCmd = &cobra.Command{
+// Bundle add command ('mixer bundle add')
+type bundleAddCmdFlags struct {
+	allLocal    bool
+	allUpstream bool
+	git         bool
+}
+
+var bundleAddFlags bundleAddCmdFlags
+
+var bundleAddCmd = &cobra.Command{
 	Use:   "add [bundle(s)]",
-	Short: "Add clr-bundles to your mix",
-	Long:  `Add clr-bundles to your mix`,
+	Short: "Add local or upstream bundles to your mix",
+	Long:  `Add local or upstream bundles to your mix`,
 	RunE: func(cmd *cobra.Command, args []string) error {
 		var bundles []string
-		if bundleFlags.all == false {
+		if bundleAddFlags.allLocal == false && bundleAddFlags.allUpstream == false {
 			if len(args) == 0 {
-				return errors.New("bundle add requires at least 1 argument if --all is not passed")
+				return errors.New("bundle add requires at least 1 argument if neither --all-local nor --all-upstream are passed")
 			}
 			bundles = strings.Split(args[0], ",")
 		}
 		b, err := builder.NewFromConfig(config)
 		if err != nil {
 			fail(err)
 		}
-		numadded, err := b.AddBundles(bundles, bundleFlags.force, bundleFlags.all, bundleFlags.git)
-		fmt.Println(numadded, " bundles were added")
+		err = b.AddBundles(bundles, bundleAddFlags.allLocal, bundleAddFlags.allUpstream, bundleAddFlags.git)
+		if err != nil {
+			fail(err)
+		}
+		return nil
+	},
+}
+
+// Bundle list command ('mixer bundle list')
+type bundleListCmdFlags struct {
+	tree bool
+}
+
+var bundleListFlags bundleListCmdFlags
+
+var bundleListCmd = &cobra.Command{
+	Use:   "list [mix|local|upstream]",
+	Short: "List bundles",
+	Long: `List either:
+  mix       The bundles in the mix (recursively following includes), or
+  local     The available local bundles, or 
+  upstream  The available upstream bundles`,
+	Args:      cobra.OnlyValidArgs,
+	ValidArgs: []string{"mix", "local", "upstream"},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return errors.New("bundle list requires 1 argument")
```

Comment:
Can this be changed and the description updated to `When no arguments are passed, mixer bundle list reports all the bundles that will be in your mix`?

Reply:
So I can type `mixer bundle list` without having to think :)

Reply:
Ah, yes. I actually meant to make "mix" the default value, and it just got lost in the shuffle.